cmd/gazctl/gazctlcmd/gazctl.go

```diff
@@ -0,0 +1,136 @@
+	})
+
+	JournalRegisterCommands []RegisterCommandFunc
+	ShardRegisterCommands   []RegisterCommandFunc
+)
+
+// Functions used to register sub-command with a parent
+type RegisterCommandFunc func(*flags.Command) error
+
+// ListConfig is common configuration of list operations.
+type ListConfig struct {
+	Selector string   `long:"selector" short:"l" description:"Label Selector query to filter on"`
+	Format   string   `long:"format" short:"o" choice:"table" choice:"yaml" choice:"json" choice:"proto" default:"table" description:"Output format"`
+	Labels   []string `long:"label-columns" short:"L" description:"Labels to present as columns, eg -L label-one -L label-two"`
+	Primary  bool     `long:"primary" short:"p" description:"Show primary column"`
+	Replicas bool     `long:"replicas" short:"r" description:"Show replicas column"`
+	RF       bool     `long:"rf" description:"Show replication factor column"`
+}
+
+// ApplyConfig is common configuration of apply operations.
+type ApplyConfig struct {
+	SpecsPath  string `long:"specs" default:"-" description:"Input specifications path to apply. Use '-' for stdin"`
+	DryRun     bool   `long:"dry-run" description:"Perform a dry-run of the apply"`
+	MaxTxnSize int    `long:"max-txn-size" default:"0" description:"maximum number of specs to be processed within an apply transaction. If 0, the default, all changes are issued in a single transaction"`
+}
+
+// EditConfig is common configuration for exit operations.
+type EditConfig struct {
+	Selector   string `long:"selector" short:"l" required:"true" description:"Label Selector query to filter on" no-ini:"true"`
+	MaxTxnSize int    `long:"max-txn-size" default:"0" description:"maximum number of specs to be processed within an apply transaction. If 0, the default, all changes are issued in a single transaction"`
+}
+
+type pruneConfig struct {
+	Selector string `long:"selector" short:"l" required:"true" description:"Label Selector query to filter on"`
+	DryRun   bool   `long:"dry-run" description:"Perform a dry-run of the apply"`
+}
+
+func (cfg ApplyConfig) decode(into interface{}) error {
+	var buffer []byte
+	var err error
+
+	if cfg.SpecsPath == "-" {
+		buffer, err = ioutil.ReadAll(os.Stdin)
+	} else {
+		buffer, err = ioutil.ReadFile(cfg.SpecsPath)
+	}
+	mbp.Must(err, "failed to read YAML input")
+
+	if err = yaml.UnmarshalStrict(buffer, into); err != nil {
+		// `yaml` produces nicely formatted error messages that are best printed as-is.
+		_, _ = os.Stderr.WriteString(err.Error() + "\n")
+		return errors.New("YAML decode failed")
+	}
+	return nil
+}
+
+func startup() {
+	mbp.InitLog(baseCfg.Log)
+	protocol.RegisterGRPCDispatcher(baseCfg.Zone)
+}
+
+func mustAddCmd(cmd *flags.Command, name, short, long string, cfg interface{}) *flags.Command {
+	cmd, err := cmd.AddCommand(name, short, long, cfg)
+	mbp.Must(err, "failed to add command")
+	return cmd
+}
+
+func Execute() {
```

Comment:
Should Execute be in the parent `gazctl` package ? This one bit doesn't seem re-useable.